aiida/repository/control.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Module for overall repository control commands."""
+# Note: these functions are not methods of `AbstractRepositoryBackend` because they need access to the orm.
+# This is because they have to go through all the nodes to gather the list of keys that AiiDA is keeping
+# track of (since they are descentralized in each node entry).
+# See the get_unreferenced_keyset function
+from aiida.manage.manager import get_manager
+from aiida.orm.implementation import Backend
+
+
+def repository_maintain(full: bool = False, backend: Backend = None, **kwargs) -> dict:
+    """Performs maintenance tasks on the repository."""
+    maintainance_report = {'database': {}, 'repository': {}, 'user_info': ''}
+
+    if backend is None:
+        backend = get_manager().get_backend()
+    repository_backend = backend.get_repository()
+
+    unreferenced_objects = get_unreferenced_keyset(aiida_backend=backend)
+
+    maintainance_report['repository']['unreferenced'] = len(unreferenced_objects)  # type: ignore
+    maintainance_report['repository']['info'] = repository_backend.get_info()  # type: ignore
+
+    repository_backend.delete_objects(list(unreferenced_objects))
+
+    # Perform the maintainance operations in the repository
+    maintainance_report['repository']['maintain'] = repository_backend.maintain(full=full, **kwargs)  # type: ignore
+
+    return maintainance_report
+
+
+def repository_info(statistics: bool = False, backend: Backend = None, **kwargs) -> dict:
+    """Returns relevant information on the repository."""
+    if backend is None:
+        backend = get_manager().get_backend()
+    repository_backend = backend.get_repository()
+    return repository_backend.get_info(statistics)
+
+
+def get_unreferenced_keyset(check_consistency: bool = True, aiida_backend: Backend = None, **kwargs) -> set:
+    """Returns the keyset of objects that exist in the repository but are not tracked by AiiDA.
+
+    This should be all the soft-deleted files.
+    """
+    from aiida import orm
+
+    if aiida_backend is None:
+        aiida_backend = get_manager().get_backend()
+
+    repository_backend = aiida_backend.get_repository()
+
+    keyset_backend = set(repository_backend.list_objects())
```

Comment:
Think it is a bit confusing to name this variable `_backend` when it is the repository. In current terminology, the backend refers at most to just the PostgreSQL db, but really to DB + repo.

Reply:
Fixed (but see other comment referencing this)

Reply:
Historically, the database was called the "backend", but this was really a misnomer. It was actually just *a* backend to our database. We then started using "backend" to mean the backend interface that stores all of AiiDA's data (for a given profile). We then also realized, rightfully so, that the repository also belongs to this. The database and repository are both part of the "backend" and they are even no longer fully independent.

So I think we should use the following terminology:

AiiDA uses a data store (or storage). The current backend for this consists of a database (Postgres) and a file repository (disk object store).

In the implementation we refer to it just as the "Backend" (as of @chrisjsewell 's recent changes). I think this is still a bit of a shortcut, because a backend is too general, but for internal purposes, I think it is fine.

Reply:
> AiiDA uses a data store (or storage). The current backend for this consists of a database (Postgres) and a file repository (disk object store).

Should we maybe then reflect this in the code structure? i.e.: re-arrange the stuff from `aiida.backend` and `aiida.repository` into `aiida.storage.database` and `aiida.storage.repository`? (or even `aiida.backend.database` and `aiida.backend.repository`)